src/libtcod/renderer_xterm.c

```diff
@@ -36,20 +36,31 @@
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 
 #ifdef _WIN32
 #include <windows.h>
+#else
+#include <termios.h>
+#include <unistd.h>
+#include <signal.h>
 #endif
 
 #include "error.h"
 
 #ifdef _WIN32
 static DWORD g_old_mode_stdin = 0;
 static DWORD g_old_mode_stdout = 0;
+#else
+static struct termios g_old_termios;
+static int g_got_rows = 0;
+static int g_got_columns = 0;
+static Uint32 g_got_size_timestamp = 0;
 #endif
```

Comment:
Only `termios` seems to be platform exclusive.  The other variables should exist on all platforms.
```suggestion
#endif
static int g_got_rows = 0;
static int g_got_columns = 0;
static Uint32 g_got_size_timestamp = 0;
```

Reply:
Good point, and then I've also enabled `recommended console size` for windows.